### PR TITLE
CRUD repository variable

### PR DESCRIPTION
**Pull request by garridobarrera**
# Description

Methods to create and update repository variables:

* Create repository variable: https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#create-a-repository-variable
* Update repository variable: https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#update-a-repository-variable 
* Delete repository variable: https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#delete-a-repository-variable

# Before submitting a PR:

- [x] Changes must not break binary backwards compatibility. If you are unclear on how to make the change you think is needed while maintaining backward compatibility, [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Add JavaDocs and other comments as appropriate. Consider including links in comments to relevant documentation on https://docs.github.com/en/rest .  
- [x] Add tests that cover any added or changed code. This generally requires capturing snapshot test data. See [CONTRIBUTING.md](CONTRIBUTING.md) for details.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.
- [x] Push your changes to a branch other than `main`. You will create your PR from that branch.

# When creating a PR: 

- [x] Fill in the "Description" above with clear summary of the changes. This includes:
  - [x] If this PR fixes one or more issues, include "Fixes #<issue number>" lines for each issue. 
  - [x] Provide links to relevant documentation on https://docs.github.com/en/rest where possible.
- [x] All lines of new code should be covered by tests as reported by code coverage. Any lines that are not covered must have PR comments explaining why they cannot be covered. For example, "Reaching this particular exception is hard and is not a particular common scenario."
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by garridobarrera** [comment]
@bitwiseman I had a problem with this PR https://github.com/hub4j/github-api/pull/1668 and close it by error so i have opened this one.

File: src/test/java/org/kohsuke/github/GHRepositoryTest.java [hunk 1]
```diff
@@ -1610,4 +1657,5 @@ public void cannotRetrievePermissionMaintainUser() throws IOException {
         GHPermissionType permission = r.getPermission("alecharp");
         assertThat(permission.toString(), is("UNKNOWN"));
     }
+
```
Comment (bitwiseman) [inline src/test/java/org/kohsuke/github/GHRepositoryTest.java:1660]:
```suggestion
```

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 2]
```diff
@@ -2738,10 +2752,22 @@ public GHContent getReadme() throws IOException {
      * @throws IOException
      *             the io exception
      */
+    @Deprecated
     public GHRepositoryVariable getRepoVariable(String name) throws IOException {
-        return root().createRequest()
-                .withUrlPath(getApiTailUrl("actions/variables"), name)
-                .fetch(GHRepositoryVariable.class);
+        return getVariable(String name);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepository.java:2757]:
```suggestion
        return getVariable(name);
```

**Review by bitwiseman — APPROVED** [review]
@garridobarrera 
Great work! Thanks!

I made some minor tweaks.  Will merge when CI passes.

**Comment by dougj-contrast** [comment]
Hello @bitwiseman is there a process or request that can be made to get this change rolled into a new release? We would like to take advantage of this new feature, but noticed it was merged after the last release cut 06/01.

Please advise or let me know if we should open an issue requesting a new release, thanks!

**Pull request merged**